src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -35,6 +40,64 @@ public IndexModel createIndex(CreateIndexRequest createIndexRequest) throws Pine
         return indexModel;
     }
 
+    public <E extends Enum<E>> boolean validateEnums(String stringToValidate, List<E> validEnums) {
+        return validEnums.stream()
+                .anyMatch(validEnum -> Objects.equals(stringToValidate, validEnum.toString()));
+    }
+
+    public IndexModel createServerlessIndex(String indexName, String metric, int dimension, String cloud,
+                                            String region) {
+        if (indexName == null || indexName.isEmpty()) {
+            throw new PineconeValidationException("Index name cannot be null or empty");
+        }
+
+        if (metric == null || metric.isEmpty()) {
+            throw new PineconeValidationException("Metric cannot be null or empty. Must be 'euclidean', 'cosine' or 'dot-product'");
+        }
+        List<IndexMetric> indexMetricEnums = Arrays.asList(IndexMetric.values());
+        if (!validateEnums(metric, indexMetricEnums)) {
+            throw new PineconeValidationException(String.format("Metric must be one of %s", IndexMetric.values()));
```

Comment:
I'm not sure this will stringify the contents of the array properly. You might need to use `Arrays.toString`:

```java
......... String.format("Metric must be one of %s", Arrays.toString(IndexMetric.values())) ...........
```

Reply:
https://github.com/pinecone-io/pinecone-java-client/pull/94/commits/b3b69572d6295e13cc53e471e18eacd3fb0309a5